### PR TITLE
Create patchwork-api-base

### PR DESCRIPTION
**Pull request by TheGlitch76**
This module does two things:
- Keeps all our outside dependency declarations in one place
- Provides no-overhead classes Forge mods and Patchwork modules might need (i.e. constants, EnvType, etc. that doesn't really belong in the FML module)

Currently only the first is done to keep noise down.
Note git got confused when I was moving some icons that were not in the right folders.

### CODE REVIEW

**Comment by kitlith** [comment]
Hm. I have mixed feelings about this.

On the one hand, I understand why you want to do something like this. On the other hand, it might be nice to eventually specify which modules of fabric api each of our patchwork modules depends on.

But hey, we can always change it again later, and I guess it's not like patchwork's modules are actually meant to be used independently anyway.

**Comment by TheGlitch76** [comment]
"If we include the whole API we're going to use the whole API"
in other words: Until some actually decent solution of using individual modules exists, we may as well act like every other fabric mod here.

**Comment by rikka0w0** [comment]
Should we put some commonly used code here, like the `public static final String` for method/field signatures?

**Comment by TheGlitch76** [comment]
method/field signatures tend to be specific to one module, it's easier to keep them there. Also avoids merge conflicts (see god classes)

File: patchwork-capabilities/build.gradle [hunk 1]
```diff
@@ -2,5 +2,6 @@ archivesBaseName = "patchwork-capabilities"
 version = getSubprojectVersion(project, "0.2.0")
 
 dependencies {
-    compile project(path: ':patchwork-fml', configuration: 'dev')
+	implementation project(path: ':patchwork-api-base', configuration: 'dev')
```
Comment (Rongmario) [inline patchwork-capabilities/build.gradle:5]: Mixed tabs + spaces

**Comment by kitlith** [comment]
^ if we're going to make everything depend on base anyway, then having them all share the single icon might be nice.

**Comment by TheGlitch76** [comment]
forgot to put it here: sharing icons isn't currently possible

**Pull request merged**